modal-go/test/sandbox_test.go

```diff
@@ -832,3 +832,201 @@ func TestSandboxExperimentalDockerMock(t *testing.T) {
+	g.Expect(string(output1)).To(gomega.Equal("hello\n"))
+
+	output2, err := io.ReadAll(p.Stdout)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	g.Expect(string(output2)).To(gomega.Equal(""))
+
+	exitCode, err := p.Wait(ctx)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	g.Expect(exitCode).To(gomega.Equal(0))
+}
+
+func TestSandboxExecBinaryMode(t *testing.T) {
+	t.Parallel()
+	g := gomega.NewWithT(t)
+	ctx := context.Background()
+	tc := newTestClient(t)
+
+	app, err := tc.Apps.FromName(ctx, "libmodal-test", &modal.AppFromNameParams{CreateIfMissing: true})
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	image := tc.Images.FromRegistry("alpine:3.21", nil)
+
+	sb, err := tc.Sandboxes.Create(ctx, app, image, nil)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	defer terminateSandbox(g, sb)
+
+	p, err := sb.Exec(ctx, []string{"printf", "\\x01\\x02\\x03"}, nil)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	output, err := io.ReadAll(p.Stdout)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	g.Expect(output).To(gomega.Equal([]byte{0x01, 0x02, 0x03}))
+
+	exitCode, err := p.Wait(ctx)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	g.Expect(exitCode).To(gomega.Equal(0))
+}
+
+func TestSandboxExecWithPty(t *testing.T) {
+	t.Parallel()
+	g := gomega.NewWithT(t)
+	ctx := context.Background()
+	tc := newTestClient(t)
+
+	app, err := tc.Apps.FromName(ctx, "libmodal-test", &modal.AppFromNameParams{CreateIfMissing: true})
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	image := tc.Images.FromRegistry("alpine:3.21", nil)
+
+	sb, err := tc.Sandboxes.Create(ctx, app, image, nil)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	defer terminateSandbox(g, sb)
+
+	p, err := sb.Exec(ctx, []string{"echo", "hello"}, &modal.SandboxExecParams{PTY: true})
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	exitCode, err := p.Wait(ctx)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	g.Expect(exitCode).To(gomega.Equal(0))
+}
+
+func TestSandboxExecWaitTimeout(t *testing.T) {
+	t.Parallel()
+	g := gomega.NewWithT(t)
+	ctx := context.Background()
+	tc := newTestClient(t)
+
+	app, err := tc.Apps.FromName(ctx, "libmodal-test", &modal.AppFromNameParams{CreateIfMissing: true})
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	image := tc.Images.FromRegistry("alpine:3.21", nil)
+
+	sb, err := tc.Sandboxes.Create(ctx, app, image, nil)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	defer terminateSandbox(g, sb)
+
+	p, err := sb.Exec(ctx, []string{"sleep", "999"}, &modal.SandboxExecParams{Timeout: 1 * time.Second})
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	t0 := time.Now()
+	exitCode, err := p.Wait(ctx)
+	elapsed := time.Since(t0)
+
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	g.Expect(elapsed).To(gomega.BeNumerically(">", 800*time.Millisecond))
+	g.Expect(elapsed).To(gomega.BeNumerically("<", 10*time.Second))
+	g.Expect(exitCode).To(gomega.Equal(0))
+}
+
+func TestSandboxExecOutputTimeout(t *testing.T) {
```

Comment:
When running off this PR, I sometimes get this error:

```
=== CONT  TestSandboxExecOutputTimeout
    sandbox_test.go:1026: 
        Unexpected error:
            <modal.ExecTimeoutError>: 
            ExecTimeoutError: deadline exceeded while waiting for exec 9c0c5a58-4ee6-41ba-970a-a33800af637f
            {
                Exception: "deadline exceeded while waiting for exec 9c0c5a58-4ee6-41ba-970a-a33800af637f",
            }
        occurred
--- FAIL: TestSandboxExecOutputTimeout (2.79s)
```

You can reproduce running: `go test -v -count=10  ./test -run TestSandboxExecOutputTimeout`